gtsam/hybrid/HybridGaussianFactorGraph.h

```diff
@@ -220,34 +220,89 @@ class GTSAM_EXPORT HybridGaussianFactorGraph
    * @brief Compute the VectorValues solution for the continuous variables for
    * each mode.
    *
+   * @tparam BAYES Template on the type of Bayes graph, either a bayes net or a
+   * bayes tree.
    * @param discrete_keys The discrete keys which form all the modes.
-   * @param continuousBayesNet The Bayes Net representing the continuous
+   * @param continuousBayesNet The Bayes Net/Tree representing the continuous
    * eliminated variables.
    * @param assignments List of all discrete assignments to create the final
    * decision tree.
    * @return DecisionTree<Key, VectorValues::shared_ptr>
    */
+  template <typename BAYES>
   DecisionTree<Key, VectorValues::shared_ptr> continuousDelta(
       const DiscreteKeys& discrete_keys,
-      const boost::shared_ptr<BayesNetType>& continuousBayesNet,
-      const std::vector<DiscreteValues>& assignments) const;
+      const boost::shared_ptr<BAYES>& continuousBayesNet,
+      const std::vector<DiscreteValues>& assignments) const {
+    // Create a decision tree of all the different VectorValues
+    std::vector<VectorValues::shared_ptr> vector_values;
+    for (const DiscreteValues& assignment : assignments) {
+      VectorValues values = continuousBayesNet->optimize(assignment);
+      vector_values.push_back(boost::make_shared<VectorValues>(values));
+    }
+    DecisionTree<Key, VectorValues::shared_ptr> delta_tree(discrete_keys,
+                                                           vector_values);
+
+    return delta_tree;
+  }
 
   /**
    * @brief Compute the unnormalized probabilities of the continuous variables
    * for each of the modes.
    *
+   * @tparam BAYES  Template on the type of Bayes graph, either a bayes net or a
+   * bayes tree.
    * @param discrete_keys The discrete keys which form all the modes.
    * @param continuousBayesNet The Bayes Net representing the continuous
    * eliminated variables.
    * @return AlgebraicDecisionTree<Key>
    */
+  template <typename BAYES>
   AlgebraicDecisionTree<Key> continuousProbPrimes(
```

Comment:
This is not a great name. It refers to math that is not in the documentation. Suggest `unnormalizedProbabilities` documenting the fact that it is `exp(-E)`. On a side note: I question the wisdom of exponentiating small numbers before they will somehow be normalized. Typically we avoid doing this until we know the minimum error and then subtract the minimum error so the corresponding maximum prob' == 1.0.

Reply:
These are unnormalized probabilities only on the continuous variables and `probPrime` is the method name for computing the unnormalized probabilities, so it seemed to make sense. If you still think I should rename it to `unnormalizedProbabilities`, will gladly do so.

Regarding the exponentiation, the minimum error term will always be 0 due to pruning.

Reply:
Update: this function will get deleted.